File: Residential_Building_Identification/readanssiplay.cpp
// This code will open the GeoJSON file, read the contents
// into a string, and then parse the string into a RapidJSON
// document object. It will then iterate through the features
// in the document and check if their coordinates fall within
// the specified bounding box. If the coordinates are within
// the bounding box, the feature is added to a RapidJSON array
// of filtered features. Finally, the code writes the filtered
// features to a new GeoJSON file using the RapidJSON writer.

// Note that this code assumes that the GeoJSON file follows
// the standard format for a GeoJSON file, with a top-level 
// features property that contains an array of feature objects.
// Each feature object should have a geometry property that
// contains the coordinates for the feature.

#include <iostream>
#include <fstream>
#include "rapidjson/document.h"
#include "rapidjson/stringbuffer.h"

using namespace rapidjson;
using namespace std;

int main() {
    // Open the GeoJSON file
    ifstream infile("buildings.geojson");
    stringstream buffer;
    buffer << infile.rdbuf();
    string contents(buffer.str());

    // Parse the JSON document
    Document document;
    document.Parse(contents.c_str());

    // Iterate through the features in the GeoJSON file
    for (SizeType i = 0; i < document["features"].Size(); i++) {
        // Get the coordinates of the current feature
        const Value& coordinates = document["features"][i]["geometry"]["coordinates"];
        // Print the coordinates to the screen
        cout << coordinates[1].GetDouble() << ", " << coordinates[0].GetDouble() << endl;
    }

    return 0;
}

File: Residential_Building_Identification/readanssiplay.py

# This code will open the GeoJSON file, 
# read the contents into a Python object,
# and then iterate through the features in the file. 
# For each feature, it will extract the coordinates and print them to the screen.

# Note that this code assumes that the GeoJSON file follows the standard format 
# for a GeoJSON file, with a top-level features property that contains a list 
# of feature objects. Each feature object should have a geometry property that 
# contains the coordinates for the feature.

import json

# Open the GeoJSON file
with open('buildings.geojson', 'r') as f:
    data = json.load(f)

# Iterate through the features in the GeoJSON file
for feature in data['features']:
    # Get the coordinates of the current feature
    coordinates = feature['geometry']['coordinates']
    # Print the coordinates to the screen
    print(coordinates)
